arithmetization/src/main/java/net/consensys/linea/sequencer/txselection/selectors/MaxBlockGasTransactionSelector.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright Consensys Software Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package net.consensys.linea.sequencer.txselection.selectors;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.hyperledger.besu.datatypes.PendingTransaction;
+import org.hyperledger.besu.plugin.data.TransactionProcessingResult;
+import org.hyperledger.besu.plugin.data.TransactionSelectionResult;
+import org.hyperledger.besu.plugin.services.txselection.PluginTransactionSelector;
+
+@Slf4j
+@RequiredArgsConstructor
+public class MaxBlockGasTransactionSelector implements PluginTransactionSelector {
+
+  private final long maxGasPerBlock;
+  private long cumulativeBlockGasUsed;
+  public static String TRANSACTION_GAS_EXCEEDS_MAX_BLOCK_GAS =
+      "Gas used by transaction exceeds max block gas";
+
+  @Override
+  public TransactionSelectionResult evaluateTransactionPostProcessing(
+      final PendingTransaction pendingTransaction,
+      final TransactionProcessingResult processingResult) {
+
+    final long gasUsedByTransaction = processingResult.getEstimateGasUsedByTransaction();
+
+    if (gasUsedByTransaction > maxGasPerBlock) {
+      log.trace(
+          "Not selecting transaction, gas used {} greater than max block gas {}",
+          gasUsedByTransaction,
+          maxGasPerBlock);
+      return TransactionSelectionResult.invalid(TRANSACTION_GAS_EXCEEDS_MAX_BLOCK_GAS);
+    }
+
+    if (isTransactionExceedingMaxBlockGasLimit(gasUsedByTransaction)) {
+      log.trace(
+          "Not selecting transaction, cumulative gas used {} greater than max gas per block {}",
```

Comment:
consistency of the message - `max gas per block` vs `max block gas` on line 42. I don't have a strong preference but should match the CLI option description